osquery/tables/system/posix/ssl_cert.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <openssl/asn1.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+static void fillRow(Row& r, X509* cert) {
+#define SHORT_STR 256
+  char temp[SHORT_STR + 1];
+
+  // set certificate subject information
+  auto subject_name = X509_get_subject_name(cert);
+  auto ret =
+      X509_NAME_get_text_by_NID(subject_name, NID_commonName, temp, SHORT_STR);
+  r["issued_common_name"] = ret == -1 ? "-1" : std::string(temp);
+
+  ret = X509_NAME_get_text_by_NID(
+      subject_name, NID_organizationName, temp, SHORT_STR);
+  r["issued_organization"] = ret == -1 ? "-1" : std::string(temp);
+
+  ret = X509_NAME_get_text_by_NID(
+      subject_name, NID_organizationalUnitName, temp, SHORT_STR);
+  r["issued_organization_unit"] = ret == -1 ? "-1" : std::string(temp);
+
+  auto serial = X509_get_serialNumber(cert);
+  auto bn = ASN1_INTEGER_to_BN(serial, NULL);
+  auto dec_str = BN_bn2hex(bn);
+  r["issued_serial_number"] = dec_str == NULL ? "-1" : std::string(dec_str);
+  BN_free(bn);
+  OPENSSL_free(dec_str);
+
+  // set certificate issuer information
+  auto issuer_name = X509_get_issuer_name(cert);
+  ret = X509_NAME_get_text_by_NID(issuer_name, NID_commonName, temp, SHORT_STR);
+  r["issuer_cn"] = ret == -1 ? "-1" : std::string(temp);
+
+  ret = X509_NAME_get_text_by_NID(
+      issuer_name, NID_organizationName, temp, SHORT_STR);
+  r["issuer_organization"] = ret == -1 ? "-1" : std::string(temp);
+
+  ret = X509_NAME_get_text_by_NID(
+      issuer_name, NID_organizationalUnitName, temp, SHORT_STR);
+  r["issuer_organization_unit"] = ret == -1 ? "-1" : std::string(temp);
+
+  // set period of validity
+  auto valid_from = X509_get_notBefore(cert);
+  auto valid_to = X509_get_notAfter(cert);
+  auto b = BIO_new(BIO_s_mem());
+
+  ASN1_TIME_print(b, valid_from);
+  ret = BIO_gets(b, temp, SHORT_STR);
+  r["valid_from"] = ret == 0 ? "-1" : std::string(temp);
+
+  ASN1_TIME_print(b, valid_to);
+  ret = BIO_gets(b, temp, SHORT_STR);
+  r["valid_to"] = ret == 0 ? "-1" : std::string(temp);
+  BIO_free(b);
+
+  // set sha 256 & 1 fingerprint
+  unsigned char temp_c[SHORT_STR + 1];
+  auto digest = const_cast<EVP_MD*>(EVP_sha256());
+  unsigned len = SHORT_STR;
+  ret = X509_digest(cert, digest, temp_c, &len);
+
+  std::stringstream ss;
+  for (unsigned i = 0; i < len; i++) {
+    if (ret == 0)
```

Comment:
Rather than checking the ret value here, let's bring this up above, and avoid these code paths altogether. You could change this and the section following to look like this:
```
std::stringstream ss;
  if (ret != 0) {
    for (unsigned i = 0; i < temp_digest.size(); i++) {
      ss << std::uppercase << std::hex << std::setfill('0') << std::setw(2)
         << static_cast<unsigned>(temp_digest[i]) << ':';
    }
    r["sha256_fingerprint"] = ss.str();
  } else {
    r["sha256_fingerprint"] = "-1";
  }

  temp_digest.clear();
  digest = const_cast<EVP_MD*>(EVP_sha1());
  ret = X509_digest(cert, digest, temp_digest.data(), &len);

  if (ret != 0) {
    ss.str("");
    for (unsigned i = 0; i < temp_digest.size(); i++) {
      ss << std::uppercase << std::hex << std::setfill('0') << std::setw(2)
         << static_cast<unsigned>(temp_digest[i]) << ':';
    }

    r["sha1_fingerprint"] = ss.str();
  } else {
    r["sha1_fingerprint"] = "-1";
  }
```
You also shouldn't need the `hex_str` as we can just directly access the `ss.str()` value